dist/src/main/java/io/camunda/application/commons/metrics/JfrMetricRecorder.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.application.commons.metrics;
+
+import io.camunda.application.commons.metrics.jfr.NativeMemoryMetrics;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.Objects;
+import jdk.jfr.consumer.RecordingStream;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.event.ApplicationStartedEvent;
+import org.springframework.context.annotation.Scope;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+/**
+ * Allows registering and tracking metrics based on JFR events. Although there should be no overhead
+ * according to the JVM, we still allow disabling this via an experimental flag {@code
+ * camunda.flags.jfr.metrics = false}.
+ *
+ * <p>Defaults to being enabled.
+ *
+ * <p>NOTE: the stream is not closed, as this is a singleton component; it will live until the
+ * application is closed. Since it's not using a daemon thread, this will not prevent the
+ * application from stopping.
+ */
+@Component
+@ConditionalOnProperty(
+    prefix = "camunda.flags.jfr",
+    name = "metrics",
+    havingValue = "true",
+    matchIfMissing = true)
```

Comment:
:thinking: not sure if we should have it enabled by default, considering that enabling NMT slows down by 5-10%.

Reply:
Am I wrong or without NMT it does not add any new information that is not already available in prometheus? 
If that's the case I would keep it disabled and enable it only in the zeebe-io benchmarks.
I think NMT doesn't do anything for native memory by JNI (e.g. RocksDB) so a big chunk of it will still be unaccounted for.

Reply:
Right now, because it's only NMT, yeah, it would only export RSS which we have access to (though it's somewhat more portable, since our RSS metric right now is K8s only, and this one is deployment agnostic).

The recorder here is a base for more JFR stuff though, so I could imagine we expand it. I'm fine with disabling it until then, or just having it off by default for now anyway :shrug:

Reply:
Let's leave it off, but enable it for benchmarks :smiley: